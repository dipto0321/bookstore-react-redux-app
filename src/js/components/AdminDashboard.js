import React from 'react';
import {
  Tab, Tabs, TabList, TabPanel,
} from 'react-tabs';
import 'react-tabs/style/react-tabs.scss';
import SearchForm from './SearchForm';

const AdminDashboard = () => (
  <Tabs>
    <TabList>
      <Tab>Manage Users</Tab>
      <Tab>Manage Books</Tab>
    </TabList>
    <TabPanel>
      <h2>Users</h2>
    </TabPanel>
    <TabPanel>
      <SearchForm />
    </TabPanel>
  </Tabs>
);

export default AdminDashboard;
