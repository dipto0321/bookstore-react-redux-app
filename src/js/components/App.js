import React from 'react';
import { Route, Router, Switch } from 'react-router-dom';
import { Provider } from 'react-redux';
import { createBrowserHistory } from 'history';
import decode from 'jwt-decode';
import BooksDashboard from './BooksDashboard';
import '../../scss/main.scss';
import LoginPage from '../containers/LoginPage';
import configureStore from '../store/configureStore';
import { setCurrentUser } from '../actions/currentUser';
import { setAuthorizationToken } from '../services/api';
import withAuth from '../hocs/withAuth';

export const history = createBrowserHistory();

const store = configureStore();

if (localStorage.token) {
  const userData = decode(localStorage.token);
  setAuthorizationToken(localStorage.token);

  try {
    store.dispatch(setCurrentUser({ authenticated: true, data: userData }));
  } catch (err) {
    store.dispatch(setCurrentUser({ authenticated: false, data: null }));
  }
}

const App = () => (
  <Provider store={store}>
    <Router history={history}>
      <Switch>
        <Route path="/login" component={LoginPage} />
        <Route path="/" component={withAuth(BooksDashboard, store.getState().currentUser.authenticated)} />
      </Switch>
    </Router>
  </Provider>
);

export default App;
