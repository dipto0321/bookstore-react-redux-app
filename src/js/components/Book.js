import React from 'react';
import PropTypes from 'prop-types';

const Book = (props) => {
  const { handleRemove, book } = props;
  const { id, title, category } = book;

  return (
    <div className="card">
      <div className="card-body">
        <div className="columns">
          <div className="col">
            <h4 className="genre">{category}</h4>
            <h3 className="title">{title}</h3>
            <div className="action">
              <button type="button" onClick={() => handleRemove(id)}>
                Remove
              </button>
            </div>
          </div>
          <div className="col" />
        </div>
      </div>
    </div>
  );
};

Book.propTypes = {
  book: PropTypes.instanceOf(Object).isRequired,
  handleRemove: PropTypes.func.isRequired,
};

export default Book;
