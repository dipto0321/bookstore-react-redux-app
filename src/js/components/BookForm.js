import React, { useState } from 'react';
import { connect } from 'react-redux';
import { addBookToLibrary } from '../thunks/book';
import { switchAddBookMode } from '../actions/book';
import InputWrapper from './InputWrapper';

export const BookForm = ({ book, addBookToLibrary, switchAddBookMode }) => {
  const [title, setTitle] = useState(book ? book.title : '');
  const [subTitle, setSubTitle] = useState(book ? book.subtitle : '');
  const [authors, setAuthors] = useState(book ? book.authors.join(', ') : '');
  const [description, setDescription] = useState(book ? book.description : '');
  const [publishedDate, setPublishedDate] = useState(book ? book.publishedDate : '');
  const [thumbNail, setThumbNail] = useState(book ? book.imageLinks.smallThumbnail : '');
  const [isbn, setIsbn] = useState(book ? book.industryIdentifiers[0].identifier : '');
  const [pageCount, setPageCount] = useState(book ? book.pageCount : '');
  const [category, setCategory] = useState(book ? book.categories[0] : '');

  const handleSubmit = (e) => {
    e.preventDefault();
    addBookToLibrary({
      book: {
        title,
        subtitle: subTitle,
        authors,
        description,
        published_date: publishedDate,
        thumbnail: thumbNail,
        isbn,
        page_count: pageCount,
        category,
      },
    })
      .then(() => switchAddBookMode());
  };

  return (
    <form>
      <InputWrapper
        inputValue={title}
        labelValue="Title: "
        setInput={setTitle}
        type="text"
        inputId="title"
        error={null}
      />
      <InputWrapper
        inputValue={subTitle}
        labelValue="Sub-title: "
        setInput={setSubTitle}
        type="text"
        inputId="sub-title"
        error={null}
      />
      <InputWrapper
        inputValue={authors}
        labelValue="Authors: "
        setInput={setAuthors}
        type="text"
        inputId="authors"
        error={null}
      />
      <InputWrapper
        inputValue={category}
        labelValue="Category: "
        setInput={setCategory}
        type="text"
        inputId="category"
        error={null}
      />
      <InputWrapper
        inputValue={isbn}
        labelValue="ISBN: "
        setInput={setIsbn}
        type="text"
        inputId="isbn"
        error={null}
      />
      <InputWrapper
        inputValue={description}
        labelValue="Description: "
        setInput={setDescription}
        type="textarea"
        inputId="description"
        error={null}
      />
      <InputWrapper
        inputValue={publishedDate}
        labelValue="Published Date: "
        setInput={setPublishedDate}
        type="text"
        inputId="published-date"
        error={null}
      />
      <InputWrapper
        inputValue={pageCount}
        labelValue="Page Count: "
        setInput={setPageCount}
        type="text"
        inputId="page-count"
        error={null}
      />
      <InputWrapper
        inputValue={thumbNail}
        labelValue="Thumbnail Link: "
        setInput={setThumbNail}
        type="text"
        inputId="thumbnail"
        error={null}
      />
      <div className="btn-wrapper">
        <button
          type="submit"
          className="add-book-btn"
          onClick={handleSubmit}
        >
          Add Book To Library
        </button>
      </div>
    </form>
  );
};

const mapStateToProps = state => ({
  book: state.addBookMode.book,
});

export default connect(mapStateToProps, { addBookToLibrary, switchAddBookMode })(BookForm);
