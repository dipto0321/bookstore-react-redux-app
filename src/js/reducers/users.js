import { LIST_USERS } from '../actions/actionTypes';

export default (state = [], action) => {
  if (action.type === LIST_USERS) {
    return action.users;
  }
  return state;
};
