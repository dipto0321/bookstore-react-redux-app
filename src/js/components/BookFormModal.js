import React from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import Modal from 'react-modal';
import BookForm from './BookForm';
import { switchAddBookMode } from '../actions/book';

export const BookFormModal = ({ switchAddBookMode, addBookMode }) => (
  <Modal
    isOpen={addBookMode.on}
    ariaHideApp={false}
    onRequestClose={() => switchAddBookMode(addBookMode.book)}
    closeTimeoutMS={200}
  >
    <BookForm />
  </Modal>
);

const mapStateToProps = state => ({
  addBookMode: state.addBookMode,
});

export default connect(mapStateToProps, { switchAddBookMode })(BookFormModal);
