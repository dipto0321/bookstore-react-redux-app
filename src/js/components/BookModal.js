import React from 'react';
import { connect } from 'react-redux';
import Modal from 'react-modal';
import BookForm from './BookForm';
import Book from './Book';
import { switchAddBookMode } from '../actions/book';

export const BookModal = ({ switchAddBookMode, addBookMode, isAdmin }) => (
  <Modal
    isOpen={addBookMode.on}
    ariaHideApp={false}
    onRequestClose={() => switchAddBookMode(addBookMode.book)}
    closeTimeoutMS={200}
  >
    {isAdmin && <BookForm /> }
    {!isAdmin && addBookMode.on && <Book book={addBookMode.book} /> }
  </Modal>
);

const mapStateToProps = state => ({
  addBookMode: state.addBookMode,
});

export default connect(mapStateToProps, { switchAddBookMode })(BookModal);
