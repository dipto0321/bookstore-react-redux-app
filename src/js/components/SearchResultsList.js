import React, { useEffect } from 'react';
import PropTypes from 'prop-types';
import { connect } from 'react-redux';
import BookSearchResultPanel from './BookSearchResultPanel';
import BookModal from './BookModal';
import { searchBooks } from '../thunks/book';
import history from '../services/history';

export const SearchResultsList = ({
  searchResults,
  searchTerm,
  error,
  searchBooks,
}) => {
  const isAdmin = history.location.pathname.includes('admin');

  useEffect(() => {
    if (searchTerm !== '') searchBooks(searchTerm, isAdmin);
  }, [searchBooks, searchTerm, isAdmin]);

  return (
    <div>
      <BookModal />
      {searchResults.length > 0 && (
      <h4 className="search-result-heading">
        Search results for
        {' '}
        {searchTerm}
      </h4>
      )}
      {searchResults.map(book => (
        <BookSearchResultPanel key={book.book_id} book={book} />
      ))}
      {
        error && (
        <h4>
          No results for
          {' '}
          {searchTerm}
        </h4>
        )
      }
    </div>
  );
};

const mapStateToProps = state => ({
  searchResults: state.searchResults,
  searchTerm: state.searchTerm,
  error: state.errors ? state.errors[0] : null,
});

SearchResultsList.propTypes = {
  searchResults: PropTypes.instanceOf(Object).isRequired,
  searchTerm: PropTypes.string.isRequired,
  error: PropTypes.instanceOf(Object),
  searchBooks: PropTypes.func.isRequired,
};

SearchResultsList.defaultProps = {
  error: null,
};

export default connect(mapStateToProps, { searchBooks })(SearchResultsList);
