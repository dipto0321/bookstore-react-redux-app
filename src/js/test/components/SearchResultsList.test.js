import React from 'react';
import { shallow } from 'enzyme';
import { SearchResultsList } from '../../components/SearchResultsList';
import users from '../fixtures/books';

describe('SearchResultsList', () => {
  let listSearchResults;
  let searchBooks;

  beforeEach(() => {
    listSearchResults = jest.fn();
    searchBooks = jest.fn(() => Promise.resolve());
  });

  test('should render correctly without search result', () => {
    const wrapper = shallow(
      <SearchResultsList
        listSearchResults={listSearchResults}
        searchResults={[]}
        searchTerm=""
        searchBooks={searchBooks}
      />,
    );
    expect(wrapper.find('BookSearchResultItem')).toEqual({});
  });
  test('should render correctly with search result', () => {
    const wrapper = shallow(
      <SearchResultsList
        listSearchResults={listSearchResults}
        searchResults={users}
        searchBooks={searchBooks}
        searchTerm="some term"
      />,
    );
    expect(wrapper.find('BookSearchResultItem')).toBeTruthy();
    expect(wrapper).toMatchSnapshot();
  });
});
