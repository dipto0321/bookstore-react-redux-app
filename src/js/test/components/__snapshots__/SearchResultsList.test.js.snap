// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchResultsList should render correctly with search result 1`] = `
<div>
  <ConnectFunction />
  <h4
    className="search-result-heading"
  >
    Search results for
     
    some term
  </h4>
  <ConnectFunction
    book={
      Object {
        "authors": Array [
          "Tolkien",
        ],
        "book_id": "0",
        "category": "Action",
        "chapters": "35",
        "currentChapter": "",
        "description": "Nostrud aliquip velit eu ea amet qui exercitation anim mollit do minim commodo pariatur.",
        "id": "0",
        "imageLinks": Object {
          "smallThumbnail": "",
        },
        "item_id": "zero",
        "title": "Lord of the Rings",
      }
    }
    key="0"
  />
  <ConnectFunction
    book={
      Object {
        "authors": Array [
          "Kiyosaki",
        ],
        "book_id": "1",
        "category": "Learning",
        "chapters": "24",
        "currentChapter": "",
        "description": "Nostrud aliquip velit eu ea amet qui exercitation anim mollit do minim commodo pariatur.",
        "id": "1",
        "imageLinks": Object {
          "smallThumbnail": "",
        },
        "item_id": "one",
        "title": "Guide to Investing",
      }
    }
    key="1"
  />
  <ConnectFunction
    book={
      Object {
        "authors": Array [
          "Roddenberry",
        ],
        "book_id": "2",
        "category": "Sci-Fi",
        "chapters": "30",
        "currentChapter": "",
        "description": "Nostrud aliquip velit eu ea amet qui exercitation anim mollit do minim commodo pariatur.",
        "id": "2",
        "imageLinks": Object {
          "smallThumbnail": "",
        },
        "item_id": "two",
        "title": "Star Trek",
      }
    }
    key="2"
  />
</div>
`;
